appservice/src/deploy/wizard/StopAppBeforeDeployExecuteStep.ts

```diff
@@ -3,20 +3,29 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { AzureWizardExecuteStep } from "@microsoft/vscode-azext-utils";
+import { AzureWizardExecuteStepWithActivityOutput } from "@microsoft/vscode-azext-utils";
 import { Progress, l10n } from "vscode";
-import { ext } from "../../extensionVariables";
 import { InnerDeployContext } from "../IDeployContext";
 
-export class StopAppBeforeDeployExecuteStep extends AzureWizardExecuteStep<InnerDeployContext> {
+export class StopAppBeforeDeployExecuteStep extends AzureWizardExecuteStepWithActivityOutput<InnerDeployContext> {
+    stepName: string = 'StopAppBeforeDeployExecuteStep';
+    protected getTreeItemLabel(context: InnerDeployContext): string {
+        return l10n.t('Stop app "{0}" before deployment', context.site.fullName);
+    }
+    protected getOutputLogSuccess(context: InnerDeployContext): string {
+        return l10n.t('Successfully stopped app "{0}".', context.site.fullName);
+    }
+    protected getOutputLogFail(context: InnerDeployContext): string {
+        return l10n.t('Failed to stop app "{0}".', context.site.fullName);
+    }
+    protected getOutputLogProgress(context: InnerDeployContext): string {
+        return l10n.t('Stopping app "{0}"...', context.site.fullName);
+    }
+
     public priority: number = 100;
-    public async execute(context: InnerDeployContext, progress: Progress<{ message?: string; increment?: number }>): Promise<void> {
+    public async execute(context: InnerDeployContext, _progress: Progress<{ message?: string; increment?: number }>): Promise<void> {
         const site = context.site;
         const client = await site.createClient(context);
-
-        const stoppingApp = l10n.t('Stopping app...');
-        progress.report({ message: stoppingApp });
```

Comment:
I'll let you make the final call, but there's a couple reasons we might still want to include progress reports:

1. I think it fires the `onProgress` event and refreshes the activity tree item
2. For users of the wizard that aren't using activities, it provides a message to show in the `vscode.window.showProgress` box